### PR TITLE
Fix root ID calculation when check if block can be transformed

### LINKED ISSUE #60026: Potential regression in TemplateLock behavior.
### Description

I've received some information (DM) that 6.5 may have introduced a regression in the TemplateLock behavior. 

### Step-by-step reproduction instructions

The conditions are supposed to be the following:

 - A container with "templateLock=all" (can be top level `wp.data.dispatch( 'core/block-editor' ).updateSettings( { templateLock: 'all' } )` )
 - An unlocked nested container within that locked container.
 - You should be able to transform blocks that are within the nested container but apparently it's not possible anymore in 6.5.

Unfortunately, I don't have more information than that. This needs to be investigated to confirm whether there's a real regression or not.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

### PR DESCRIPTION
**Pull request by jsnajdr**
Fixes #60026. Fixing a regression introduced in #57892.

A `clientIds` array is passed to `getBlockRootClientId`, but the selector expects only one `clientId`. It leads to returning `null`, which means the root of the document.

Then a `canRemoveBlocks` permission check returns a wrong result, because it misses an intermediate "Group" block in the hierarchy that resets `templateLock` from the root `'all'` to `false`.

One thing I'm not happy about is that we're checking the root ID only of the first block in a multi-selection. That could lead to a wrong behavior where blocks are transformed even if the second and later blocks in the selection are not transformable/removable.

### CODE REVIEW

**Review by mcsf — APPROVED** [review]
Nice work.

**Comment by getdave** [comment]
This PR was triaged by Editor release leads and contributors in WP Slack as part of the final WordPress 6.5 process.

See https://wordpress.slack.com/archives/C02QB2JS7/p1711371488652689 (requires registration).

The criteria used was:

- Was this bug identified during the RC cycle for WP 6.5?
- Is this bug directly related to one of the blessed features in this release?
- Would this bug cause existing website’s to break/fail upon upgrade to WP 6.5?
- What is the approximate scope of the impact? Scale: (Minimal) 1 - 5 (Severe).
- Is the fix introducing any new features or changes?


**Conclusion**: we propose that this fix _is_ included in WP 6.5.

**Reason**: change surface area is _very_ small and impact is valuable.

We now await a final decision from the full release team.

**Comment by ntsekouras** [comment]
Thanks so much for handling this!

**Comment by bph** [comment]
I just cherry-picked this PR to the release/18.0 branch to get it included in the next release: 01f32c849f

**Comment by getdave** [comment]
As part of a triage by release leads it was requested that tests be added to provide coverage for this PR.

See https://wordpress.slack.com/archives/C02RQBWTW/p1711552210253759

**Comment by getdave** [comment]
I just cherry picked this into WP 6.5 RC 4 branch https://github.com/WordPress/gutenberg/pull/60241/commits/8d45734c4396169adf045f1c87b47c3c9dce28a7

**Pull request merged**
